### ISSUE TITLE
GPUParticle3D stops rendering in editor under specific circumstances

### ISSUE THREAD

**Issue by AtlaStar** [godotengine/godot#97680]
### Tested versions

4.4.dev

### System information

Godot v4.4.dev (76a135926) - Windows 10.0.19045 - Multi-window, 1 monitor - Vulkan (Forward+) - dedicated AMD Radeon RX 6650 XT (Advanced Micro Devices, Inc.; 31.0.24033.1003) - AMD Ryzen 7 3800X 8-Core Processor (16 threads)

### Issue description

When the Fixed FPS is set to 0 for a GPUParticle3D, and the material assigned to it's mesh has billboarding with the keep scale flag enabled, new particles are no longer emitted. Changing between nodes will briefly cause a burst of particles. New particles also seem to spawn while the editor is minimized.

Is related to #97621 as the default for a CPUParticle3D's Fixed FPS is 0, so on swapping to a GPUParticle3D, and having a mesh material that has scaled billboarding enabled, the already emitted particles will render as normal but no new particles will be added to the system, which may briefly appear as if the wrong amount of particles are being generated.

### Steps to reproduce

As described above:

- Make a GPUParticle3D node.
- set a quadmesh and create a material for it
- have that material be set to use any of the billboarding options
- enable the keep scale flag.

### Minimal reproduction project (MRP)

I have been using the same MRP as was submitted on #97621, which I will include here again.
[New.Compressed.zipped.Folder.zip](https://github.com/user-attachments/files/17201033/New.Compressed.zipped.Folder.zip)

**Comment by AtlaStar** [comment]
Tried to find a possible solution, but as I am not familiar with the render system I was unable to find the offending area of concern, but the fact materials and particles are both involved, my hunch is possibly a shader is to blame.

**Comment by AtlaStar** [comment]
Also forgot to note that attaching a visual studio debugger and breaking inside of the particles_storage class showed that the particle count was correct in functions pertaining to setting up the buffers buffers for the render device, which is what led me to believe that the issue exists somewhere in shader land in addition to the fact that certain material properties are required to cause the improper behavior.

**Comment by AtlaStar** [comment]
More weird behavior discovered; hovering over _any_  GUI element that can update to display a tooltip will briefly cause the particle node to burst some particles while it is in the errored state of not drawing particles.

**Comment by AtlaStar** [comment]
Making as a separate comment because it is relevant; the original test used as a ParticleProcessMaterial that has a curve assigned to its display scale. This plus a Fixed FPS of 0, and billboarding for the material set to keep scale are all required to display the behavior described

**Comment by AtlaStar** [comment]
Alright, so I found that something about the previously mentioned series of events is causing the M11 minor of each particles transformation matrix to all be set to NaN

**Comment by FireCatMagic** [comment]
Thanks for all of this, it seems like you found the real issue that was causing my issue

**Comment by AtlaStar** [comment]
Yeah, it is a tricky one and basically a perfect storm of multiple things that are all required in order to cause the issue. Trying to find a more permanent fix but having trouble figuring out where the transform data is being modified to NaN since the buffer where those values exist is opaque in the visual studio profiler and only shows as an ID.

**Comment by AtlaStar** [comment]
@FireCatMagic pretty sure I found a fix, took me way longer than it should have to find the cause lol. Gonna test more to validate things work as expected then push the fix
